File: Programmers/Stack/기능개발.cpp
#include <string>
#include <vector>

using namespace std;

vector<int> solution(vector<int> progresses, vector<int> speeds) {
    vector<int> answer;
    vector<int> days;
    for (int i = 0; i < progresses.size(); i++) {
        int remain = 100 - progresses[i];
        int day = (remain + speeds[i] - 1) / speeds[i];
        days.push_back(day);
    }

    int cur = days[0];
    int cnt = 0;

    for (int i = 0; i < days.size(); i++) {
        if (days[i] <= cur) {
            cnt++;
        } else {
            answer.push_back(cnt);
            cnt = 1;
            cur = days[i];
        }
    }

    answer.push_back(cnt);

    return answer;
}

File: Programmers/Stack/기능개발.py
from collections import deque


def solution(progresses, speeds):
    answer = []
    queue = deque()  # 작업 완료 일수

    for p, s in zip(progresses, speeds):
        queue.append((100 - p + s - 1) // s)  # 올림

    cur = queue.popleft()
    cnt = 1

    while queue:
        day = queue.popleft()
        if day <= cur:
            cnt += 1
        else:
            answer.append(cnt)
            cur = day
            cnt = 1

    answer.append(cnt)  # 마지막 추가

    return answer